src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3496,7 +3496,9 @@ class CreateStatementSegment(BaseSegment):
             Sequence(
                 "NOTIFICATION_PROVIDER",
                 Ref("EqualsSegment"),
-                OneOf("AWS_SNS", "AZURE_EVENT_GRID", "GCP_PUBSUB"),
+                OneOf(
+                    "AWS_SNS", "AZURE_EVENT_GRID", "GCP_PUBSUB", "AZURE_STORAGE_QUEUE"
+                ),
```

Comment:
Do you want to add a quoted option here too, as per my change request below?

Reply:
Should now be resolved.